File: 1_Two_Sum/hash_table.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash_num2idx = {}
        for i in range(len(nums)):
            value = hash_num2idx.get(target-nums[i])
            if value != None:
                return i, value
            hash_num2idx[nums[i]] = i

# Time Complexity: O(n)
# Runtime: 122 ms	Memory: 15.2 MB

File: 1_Two_Sum/hash_table.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int remainder;
        vector<int> indices;
        unordered_map<int, int> umap;
        for (int i = 0; i < nums.size(); i++){
            remainder = target - nums[i];
            if (umap.count(remainder)){
                indices.push_back(i);
                indices.push_back(umap[remainder]);
                return indices;
            }
            umap[nums[i]] = i;
        }
        return indices;
    }
};

// Time Complexity: O(n)
// Runtime: 8 ms	Memory: 10.8 MB